Lib/test/test_traceback.py

```diff
@@ -3802,6 +3802,45 @@ def test_traceback_header(self):
         exc = traceback.TracebackException(Exception, Exception("haven"), None)
         self.assertEqual(list(exc.format()), ["Exception: haven\n"])
 
+    def test_name_error_punctuation_with_suggestions(self):
+        def raise_mssage(message, name):
+            try:
+                raise NameError(message, name=name)
+            except NameError as e:
+                exc = traceback.TracebackException.from_exception(e)
+                return list(exc.format())[-1]
+
+        test_cases = [
+            ("a.", "time", "NameError: a. Did you forget to import 'time'?\n"),
```

Comment:
Please duplicate the NameError from the IDLE tests here where we test "Or did you forget?".